fitbenchmarking/fitbenchmarking
# Introduce standard colourmaps for table shading

#### Description of Work

Draft to address #832. The  `colour_scale` plotting option has been replaced with three new parameters: `colour_map`, `colour_ulim` and `cmap_range`. The text colour in the table has been set to be white. The `colour_map` string now matches the name of a selected standard colourmap from matplotlib, the choice has been narrowed down to sequential colour maps that are visible with **white text**, valid options are: 

`'colour_map': ['viridis', 'plasma', 'inferno', 'magma', 'cividis',
                        'Greys', 'Purples', 'Blues', 'Greens', 'Oranges', 'Reds',
                        'YlOrBr', 'YlOrRd', 'OrRd', 'PuRd', 'RdPu', 'BuPu',
                        'GnBu', 'PuBu', 'YlGnBu', 'PuBuGn', 'BuGn', 'YlGn']`
![image](https://user-images.githubusercontent.com/75368443/121561190-2ab93380-ca10-11eb-9047-abc77e321a1e.png)


![image](https://user-images.githubusercontent.com/75368443/121561167-242abc00-ca10-11eb-9d8a-98abf8ec4570.png)


Colours are then sampled from the colourmap according to the relative values (normalised in a log scale) and converted to a hex format and stored in a dictionary. The `colour_ulim` defines the value at which the colour will not change, so that an extremely relative high values do not distort the scale; this defaults to a value of 100. The `cmap_range: [low, high]` option allows a custom range  within the colourmap to be sampled from, avoiding the extreme colours of the scales which can improve aesthetics of table and ensure readability.

#### Testing Instructions

1. Run fitbenchmarking and check that table colours are readable ( default `'magma'` )
2. Experiment with different colour maps and values for `colour_ulim` and `cmap_range`

#### Documentation: Is there a suitable change to documentation for this change?

If we are happy with this change, the documentation will need to be updated.
